### DISCUSSION TITLE
Is there any other version of this project instead of nerfstudio?

### DISCUSSION THREAD

**Discussion by JerryPW** [lsongx/nerfplayer-nerfstudio#12] [Q&A] [answered]
Hi! Thanks for your great job! I've been using NeRF in long video 3D reconstruction and found your work amazing. 

However, it seems that the code is not complete, some important parts like deformation field network has not been found in your provided code. Furthermore, the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio, thus seems inconvenient. Therefore, I wonder is there any other version or this project can be provided? 

Hoping for your reply soon!

**Comment by lsongx** [comment]
Hi @JerryPW , yes there is another version and I can share you through email.
For your question:
> the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio

I'm not entirely sure what you meant. Do you mind suggesting me some examples?

**Comment by JerryPW** [comment]
> Hi @JerryPW , yes there is another version and I can share you through email. For your question:
> 
> > the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio
> 
> I'm not entirely sure what you meant. Do you mind suggesting me some examples?

Thanks for reply! Here's my email: li-hao@sjtu.edu.cn
As for your question, I could put it another way: if I want to use ```ns-process-data```command in nerfstudio to prepare my dataset, which provides 'images', 'video', 'polycam', 'record3d'..... and any other type of version, which type of dataset should I choose?

**Comment by lsongx** [comment]
You can prepare your dataset by following https://github.com/KAIR-BAIR/dycheck/blob/main/docs/RECORD3D_CAPTURE.md

**Comment by ElsayedMMohammed** [comment]
@Isongx

> Hi @JerryPW , yes there is another version and I can share you through email. For your question:
> 
> > the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio
> 
> I'm not entirely sure what you meant. Do you mind suggesting me some examples?

Hi! Can you please share it with me as well? I'm working on DNeRF idea and your code will be greatly useful. Thank you. My email is elsayed.mohammed@ucalgary.ca

**Comment by xuyaojian123** [comment]
> Hi @JerryPW , yes there is another version and I can share you through email. For your question:
> 
> > the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio
> 
> I'm not entirely sure what you meant. Do you mind suggesting me some examples?

Hi! Can you please share it with me as well? My email is 2450703254@qq.com. Thank your very much!!!

**Comment by xuyaojian123** [comment]
> > Hi @JerryPW , yes there is another version and I can share you through email. For your question:
> > > the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio
> > 
> > 
> > I'm not entirely sure what you meant. Do you mind suggesting me some examples?
> 
> Thanks for reply! Here's my email: [li-hao@sjtu.edu.cn](mailto:li-hao@sjtu.edu.cn) As for your question, I could put it another way: if I want to use `ns-process-data`command in nerfstudio to prepare my dataset, which provides 'images', 'video', 'polycam', 'record3d'..... and any other type of version, which type of dataset should I choose?

Hi! Can you please share it with me as well? My email is [2450703254@qq.com](mailto:2450703254@qq.com). Thank your very much!!!

**Comment by zdwww** [comment]
> Hi @JerryPW , yes there is another version and I can share you through email. For your question:
> 
> > the prepared dataset seems not compatible with the standard version dataset provided in nerfstudio
> 
> I'm not entirely sure what you meant. Do you mind suggesting me some examples?

Hi there! Do you mind also share the other version of Nerfplayer with me? My email is zdw199948@icloud.com. Thank you so much! Nerfplayer is such an exciting work!

**Comment by Friedrich-M** [comment]
@lsongx Hi, could you kindly share the complete version of NeRFPlayer with me? My email address is moulz1031@gmail.com. Thank you for your great work and your attention!

**Comment by achao2013** [comment]
Exciting work! Could you kindly share the complete version of NeRFPlayer with me? My email address is chao_jichao@hotmail.com. Thank you very much!  @lsongx

**Comment by seoha-kim** [comment]
Hello. Can I also ask you for the complete NeRFPlayer code? My email address is kim-seoha@naver.com. Thank you. @lsongx

**Comment by joaxkal** [comment]
Hi, thank you for the great work! I'd also like to ask for the code joannakaletaa@gmail.com, @lsongx

**Comment by wangshu31** [comment]
Hello @lsongx , I am very interested in your work.  I would like to learn more about the implementation, could you please share your source code with me? I would appreciate it a lot. My email is wangshu0531@outlook.com .

**Comment by origamiman72** [comment]
Hi @lsongx , I'm working on extending Nerfplayer. Could I also get the code at ahujaabhik@berkeley.edu? Thanks!

**Comment by JPaulMora** [comment]
Hello all, @lsongx thanks for your great work. Could I suggest publishing the non-nerfstudio version here on github? In any case, could I have the code mor15799@uvg.edu.gt? Thanks

**Reply by lsongx** [reply, marked as answer]
Good idea! You can find it here: https://drive.google.com/file/d/1Ps5_eux1EdQH2luDjDBfriROHYVji31a/view?usp=drive_web

For ngp based code:
The main file should be `main_stream.py`
The core implementation should be `stream/frame_grid_encoder.py`

For tensorf based code:
The main file should be `stream_train.py`
The core implementation should be `models/stream_tensoRF.py` and `models/stream_tensorBase.py`

**Reply by JPaulMora** [reply]
Woah,timely response! Cool, now say goodbye to straight 8h of tinkering with this. Thanks a lot!

**Reply by lsongx** [reply]
No problem! Feel free to post here (or start another discussion) if you need help on the previous (old and messy😅) version!

**Reply by Henryzyf** [reply]
Hello! The g-drive link above is not valid now. Could you please update the link or share me with email (henryzhouyufu@sjtu.edu.cn). Thanks!

**Comment by Pinocchio-Jia** [comment]
@lsongx Hi, I am very interested in your work on Dynamic Scene NeRF. Your work is impressive and captivating. I would like to request the complete code for NeRFPlayer implemented in ngp and tensorf, along with a portion of the dataset. My email is pinocchiojia@gmail.com.

**Reply by lsongx** [reply]
Hi, please get the code by clicking the g-drive link above.

**Reply by Pinocchio-Jia** [reply]
![image](https://github.com/lsongx/nerfplayer-nerfstudio/assets/138574367/1e83a3b9-f39e-463e-adee-f47cb7a38ae8)
![image](https://github.com/lsongx/nerfplayer-nerfstudio/assets/138574367/a778667e-b4fe-4673-aa45-9349b86d5012)

@lsongx Hi,when the decomposition field is implemented, the frame diagram in the paper shows that the inputs are (x,y,z) and t. Why is novel_feat input in the TensorF-based code？

**Reply by lsongx** [reply]
Hi, I disabled that deformation field part by default. You can turn it on, but that needs some tuning... Sorry about that. Or, you may try the NGP based model

**Comment by Theil0** [comment]
Hello @lsongx ,
Thank you for your work. I know this discussion is getting a bit old, but I am also interested in the complete code for NeRFPlayer. As the Google Drive link above does not seem to work anymore, could you please send the code to my email ([theilo.terrisse@cs-soprasteria.com](mailto:theilo.terrisse@cs-soprasteria.com))?
Thank you very much.

**Reply by lsongx** [reply]
https://drive.google.com/file/d/1cEmyn8oEYB1GlWJlyQ0TZZZc4u9DAw22/view?usp=sharing

**Reply by Theil0** [reply]
Fetched, thank you very much for the timely response!
